react-swipe.js

```diff
@@ -65,9 +65,16 @@
     shouldComponentUpdate: function (nextProps) {
       return (
         (this.props.slideToIndex !== nextProps.slideToIndex) ||
-        (typeof this.props.shouldUpdate !== 'undefined') && !this.props.shouldUpdate(nextProps)
+        (typeof this.props.shouldUpdate !== 'undefined') && !this.props.shouldUpdate(nextProps)||
+        (nextProps.children!==this.props.children)
       );
     },
+    componentWillReceiveProps:function(newProps){
+      if(this.swipe!=null){
+        this.swipe.kill();
+      }  
+      this.swipe = Swipe(this.getDOMNode(), this.props);
```

Comment:
this will potentially restart the swipe many times, even if the props haven't changed. it's going to be quite a performance hit...? not to mention what happens if someone is in the middle of a swipe when the props are received! :)